### PR TITLE
EdkRepo: Create unit tests for the clean_command.py module and modula…

### LINKED ISSUE #357: Create unit tests for the clean_command.py module and modularize its existing methods if needed
Create unit tests for the clean_command.py module that is in edkrepo\commands\clean_command.py to ensure reliable coverage of its core functionality. As part of this work, existing methods may be modularized when necessary to make the code easier to test, maintain, and extend. The goal is to make the class more modular, so its methods are easier to test individually, resulting in smaller, more focused unit tests and improving long‑term maintainability.

### PR DESCRIPTION
**Pull request by VictorMlVillarreal**
…rize its existing methods if needed

Implemented unit tests and documentation for the clean_command.py module to improve clarity, reliability, and maintainability.

Changes:
-Added unit tests for the clean_command.py module, covering the main execution paths and edge cases. 
-Created the corresponding documentation file CleanCommand_TestCases.md. 
-All new tests rely on mocked dependencies and do not perform filesystem or network operations.

Fixes: #357

### CODE REVIEW

File: edkrepo/commands/unit_tests/test_clean_command.py [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+        CleanCommand().run_command(args, {})
+
+        mock_repo.git.clean.assert_called_once_with(f=True, d=True, n=False, q=False, x=False)
+        mock_ui.print_info_msg.assert_called_once_with(self.FORCE_OUTPUT, header=False)
+
+    @patch('edkrepo.commands.clean_command.ui_functions')
+    @patch('edkrepo.commands.clean_command.Repo')
+    @patch('edkrepo.commands.clean_command.get_workspace_manifest')
+    @patch('edkrepo.commands.clean_command.get_workspace_path')
+    def test_clean_produces_no_output_print_not_called(self, mock_get_workspace_path, mock_get_workspace_manifest, mock_repo_cls, mock_ui):
```
Comment (ashedesimone) [inline edkrepo/commands/unit_tests/test_clean_command.py]: The quiet parameter is set with the following calculation:
`args.quiet and args.force`
Meaning that the following argument combinations will result in it being set to false:

- quiet = false, force = true
- quiet = false, force = false
- quiet = true, force = false

and the following where it is set to true:

- quiet = true, force = true

Please ensure that a test case exists for each combination of arguments. 

That being said while the logic is correctly evaluating this case:
- quiet = true, force = false

I do not believe the behavior is desirable as it overrides the users specific request that quiet be enabled; we should consider correcting that.

**Review by ashedesimone — CHANGES_REQUESTED** [review]

**Review by ashedesimone — CHANGES_REQUESTED** [review]
Please squash all unit test changes into a single commit and extract the fix to the quiet flag to a separate commit. I suggest making the change to the quiet flag as the first commit and then adding the unit testing as the second commit. 

When implementing changes in the behavior of commands especially when changing the behavior or command line parameters please ensure that the appropriate command reference (https://github.com/tianocore/edk2-edkrepo/tree/main/docs/command_references) is updated. 

Please link the md file documenting the test cases here: https://github.com/tianocore/edk2-edkrepo/blob/main/docs/doc_index.md
